config/routes.rb

```diff
@@ -19,6 +19,7 @@
   resources :nodes, only: [:index, :show, :destroy] do
     collection do
       post :bootstrap
+      post :assign_roles
```

Comment:
Other option is to use `PUT/PATCH` on `/nodes` endpoint. With a payload of the type:
```
{
  "hostname1": { roles: ["kubernetes-master"] },
  "hostname2": { roles: ["kubernetes-minion"] }
}
```

Or maybe `ID's` instead of `hostnames`.